processing/src/main/java/io/druid/query/topn/TopNQueryEngine.java

```diff
@@ -124,4 +125,13 @@ public TopNQueryEngine(StupidPool<ByteBuffer> bufferPool)
 
     return new TopNMapFn(query, topNAlgorithm);
   }
+
+  public static boolean canApplyExtractionInPost(TopNQuery query)
+  {
+    return query != null
```

Comment:
I don't think we need to check for null here. We can just have Precondition.checkNotNull, since none of the instances where this is called would have a null query.